### ISSUE TITLE
fleetctl status cannot find running unit shown in list-units

### ISSUE THREAD

**Issue by ChimeraCoder** [coreos/fleet#1208]
I ran into this issue when testing out a 3-machine cluster on DigitalOcean using the alpha build (which is the default on the [Digital Ocean setup guide](https://coreos.com/docs/running-coreos/cloud-providers/digitalocean/#via-the-api).

Executing the steps on the [Quickstart](https://coreos.com/docs/quickstart/) results in a seemingly-inconsistent state. 

Running `fleetctl load hello.service` and `fleetctl start hello.service` causes the service to be started. The unit is visible with `fleetctl status hello.service` on the other two machines in the cluster, but *not* on the original machine, even though it is shown in `fleetctl list-units` and `fleetctl list-unit-files` on the original machine. (`docker ps` and `systemctl status hello.service`)

While debugging this [on IRC](https://botbot.me/freenode/coreos/2015-04-30/?msg=38018609&page=9) with @robszumski, we found an error in the `cloud-config` on the Quickstart (fixed in https://github.com/coreos/docs/pull/494), but fixing that didn't solve this issue.

![screenshot - 04302015 - 04 17 31 pm](https://cloud.githubusercontent.com/assets/376414/7423655/50e8f400-ef65-11e4-8ed1-3b3af8e293dd.png)

This was fairly reproducible; wiping and reprovisioning the machines yielded it consistently for me.

Below is the cloud-config used (copied almost exactly from the Quickstart and DigitalOcean guide, except for the aforementioned fix):

````
#cloud-config

ssh_authorized_keys:
    - ssh-rsa AAAA.... #(ommitted)
coreos:
  etcd:
    # generate a new token for each unique cluster from https://discovery.etcd.io/new?size=3
    # specify the intial size of your cluster with ?size=X
    discovery: https://discovery.etcd.io/24213cbe5c58921b22607da2a9ba151e
    # multi-region deployments, multi-cloud deployments, and droplets without
    # private networking need to use $public_ipv4
    addr: $private_ipv4:4001
    peer-addr: $private_ipv4:7001
  units:
    - name: etcd.service
      command: start
    - name: fleet.service
      command: start
````

````sh
#! /bin/sh
set -e

curl --request POST "https://api.digitalocean.com/v2/droplets" \
     --header "Content-Type: application/json" \
     --header "Authorization: Bearer $TOKEN" \
     --data '{
      "region":"nyc3",
      "image":"coreos-alpha",
      "size":"512mb",
      "name":"core-1",
      "private_networking":true,
      "ssh_keys":['$SSH_KEY_ID'],
      "user_data": "'"$(cat cloud-config.yml | sed 's/"/\\"/g')"'"
}'

curl --request POST "https://api.digitalocean.com/v2/droplets" \
     --header "Content-Type: application/json" \
     --header "Authorization: Bearer $TOKEN" \
     --data '{
      "region":"nyc3",
      "image":"coreos-alpha",
      "size":"512mb",
      "name":"core-2",
      "private_networking":true,
      "ssh_keys":['$SSH_KEY_ID'],
      "user_data": "'"$(cat cloud-config.yml | sed 's/"/\\"/g')"'"
}'

curl --request POST "https://api.digitalocean.com/v2/droplets" \
     --header "Content-Type: application/json" \
     --header "Authorization: Bearer $TOKEN" \
     --data '{
      "region":"nyc3",
      "image":"coreos-alpha",
      "size":"512mb",
      "name":"core-3",
      "private_networking":true,
      "ssh_keys":['$SSH_KEY_ID'],
      "user_data": "'"$(cat cloud-config.yml | sed 's/"/\\"/g')"'"
}'

**Comment by scole-scea** [comment]
This is #1158, I believe. It should be fixed in fleet 0.9.2 (not yet in CoreOS stable).

**Comment by bcwaldon** [comment]
@scole-scea OP is using Alpha, not Stable, so I doubt this is #1158 

@ChimeraCoder Could you run `systemctl status hello.service` on every node in the cluster? The `\x22` characters are suspicious

**Comment by ChimeraCoder** [comment]
On host `core-3` (the one I used to load/start the service), I get the following, which looks correct:

````● hello.service - My Service
   Loaded: loaded (/run/fleet/units/hello.service; linked-runtime; vendor preset: disabled)
   Active: activating (start-pre) since Fri 2015-05-01 16:39:29 UTC; 30s ago
  Process: 749 ExecStartPre=/usr/bin/docker rm hello (code=exited, status=1/FAILURE)
  Process: 680 ExecStartPre=/usr/bin/docker kill hello (code=exited, status=1/FAILURE) 
  Control: 760 (docker)
   Memory: 12.8M
   CGroup: /system.slice/hello.service
           └─control
             └─760 /usr/bin/docker pull busybox
````

But on `core-1` and `core-2`, I get:
````
● hello.service
   Loaded: not-found (Reason: No such file or directory)
   Active: inactive (dead)
````

IOW, almost the same error message that `fleetctl status hello.service` gives, except without the `\x22` characters.


Interestingly, when I tried it out just now, I actually couldn't reproduce the original error - but then I realized I had skipped the first two commands of the Quickstart guide (the curl to etcd and the `docker run`), which I had done every time I reproduced this yesterday and the day before. Wiping the clusters and starting again, this time copy/pasting all the commands on the Quickstart, I reproduced it again as expected.

**Comment by shilishper** [comment]
Just hit the same issue, but with a single coreos instance running in libvirt/qemu/kvm.
CoreOS beta (681.0.0)

**Comment by EtienneBruines** [comment]
I was able to reproduce the same strange behavior on CoreOS beta (695.2.0)

Following the same Quickstart tutorial, on DigitalOcean as well. 

```
core@odilo ~ $ fleetctl list-units
UNIT			MACHINE				ACTIVE	SUB
apache@80.service	abb7cd03.../10.129.114.205	active	running
hello.service		11dbf483.../10.129.113.198	active	running
```
## apache@80.service
```
core@odilo ~ $ fleetctl status apache@80.service
● \x22apache@80.service\x22.service
   Loaded: not-found (Reason: No such file or directory)
   Active: inactive (dead)
```
```
core@odilo ~ $ systemctl status apache@80.service
● apache@80.service - Apache web server service
   Loaded: loaded (/run/fleet/units/apache@80.service; linked-runtime; vendor preset: disabled)
   Active: active (running) since Wed 2015-06-17 20:32:51 UTC; 4min 20s ago
  Process: 3918 ExecStartPre=/usr/bin/docker pull etiennebruines/apache (code=exited, status=0/SUCCESS)
  Process: 3910 ExecStartPre=/usr/bin/docker rm apache%i (code=exited, status=1/FAILURE)
  Process: 3904 ExecStartPre=/usr/bin/docker kill apache%i (code=exited, status=1/FAILURE)
 Main PID: 3936 (docker)
   CGroup: /system.slice/system-apache.slice/apache@80.service
           └─3936 /usr/bin/docker run --name apache80 -p 146.185.187.232:80:80 etiennebruines/apache /usr/sbin/apach...

Jun 17 20:32:51 odilo docker[3918]: 246be445b5d1: Pulling dependent layers
Jun 17 20:32:51 odilo docker[3918]: 428b411c28f0: Download complete
Jun 17 20:32:51 odilo docker[3918]: 435050075b3f: Download complete
Jun 17 20:32:51 odilo docker[3918]: 9fd3c8c9af32: Download complete
Jun 17 20:32:51 odilo docker[3918]: 6d4946999d4f: Download complete
Jun 17 20:32:51 odilo docker[3918]: 246be445b5d1: Download complete
Jun 17 20:32:51 odilo docker[3918]: 246be445b5d1: Download complete
Jun 17 20:32:51 odilo docker[3918]: Status: Image is up to date for etiennebruines/apache:latest
Jun 17 20:32:51 odilo systemd[1]: Started Apache web server service.
Jun 17 20:32:51 odilo docker[3936]: AH00558: apache2: Could not reliably determine the server's fully qualifie...essage
Hint: Some lines were ellipsized, use -l to show in full.
```

## hello.service
```
core@odilo ~ $ fleetctl status hello.service 
● hello.service - My Service
   Loaded: loaded (/run/fleet/units/hello.service; linked-runtime; vendor preset: disabled)
   Active: active (running) since Wed 2015-06-17 18:41:56 UTC; 1h 59min ago
  Process: 1116 ExecStartPre=/usr/bin/docker pull busybox (code=exited, status=0/SUCCESS)
  Process: 1108 ExecStartPre=/usr/bin/docker rm hello (code=exited, status=0/SUCCESS)
  Process: 1104 ExecStartPre=/usr/bin/docker kill hello (code=exited, status=0/SUCCESS)
 Main PID: 1125 (docker)
   Memory: 1.4M
   CGroup: /system.slice/hello.service
           └─1125 /usr/bin/docker run --name hello busybox /bin/sh -c while true; do echo Hello World; sleep 1; done

Jun 17 20:41:41 oddr docker[1125]: Hello World
Jun 17 20:41:42 oddr docker[1125]: Hello World
Jun 17 20:41:43 oddr docker[1125]: Hello World
Jun 17 20:41:44 oddr docker[1125]: Hello World
Jun 17 20:41:45 oddr docker[1125]: Hello World
Jun 17 20:41:46 oddr docker[1125]: Hello World
Jun 17 20:41:47 oddr docker[1125]: Hello World
Jun 17 20:41:48 oddr docker[1125]: Hello World
Jun 17 20:41:49 oddr docker[1125]: Hello World
Jun 17 20:41:50 oddr docker[1125]: Hello World
```

**Comment by EtienneBruines** [comment]
Behavior seems fixed in the master branch ... Tried to fix it myself right now, but turns out: it's already fixed.

**Comment by crawford** [comment]
@EtienneBruines It should have been "fixed" by https://github.com/coreos/fleet/pull/1252.

**Comment by EtienneBruines** [comment]
That does like something that would fix it. 

@crawford Thank you for the confirmation / update.

**Comment by crawford** [comment]
I am able to reproduce this on 0.10.1 if I attempt to get the status from any machine other than the one on which the unit is currently running.

```
core@45.55.83.102 $ fleetctl list-units
UNIT		MACHINE				ACTIVE	SUB
redis.service	0edf52f6.../45.55.83.102	active	running
```

Same machine:
```
core@45.55.83.102 $  fleetctl status redis.service
● \x22redis.service\x22.service
   Loaded: not-found (Reason: No such file or directory)
   Active: inactive (dead)
```

Different machine:
```
core@45.55.83.100 $ fleetctl status redis.service
● redis.service
   Loaded: loaded (/run/fleet/units/redis.service; linked-runtime; vendor preset: disabled)
   Active: active (running) since Wed 2015-06-17 20:59:18 ; 3min 59s ago
 Main PID: 651 (sleep)
   Memory: 120.0K
   CGroup: /system.slice/redis.service
           └─651 /usr/bin/sleep 1000000

Jun 17 20:59:18 crawford-test-2 systemd[1]: Started redis.service.
Jun 17 20:59:18 crawford-test-2 systemd[1]: Starting redis.service...
```

**Comment by EtienneBruines** [comment]
Using the current master (8275b702b2913ecf16dd7ea650cfcd5dc0b93ec8) I am *not* able to reproduce that, not even on other machines (in the same Discovery-network). 

Using the 0.10.1 release, I was.

**Comment by crawford** [comment]
This has been fixed by https://github.com/coreos/fleet/pull/1253.

**Comment by crawford** [comment]
The fix is currently available in 717.0.0 (alpha) and 681.2.0 (stable).
